include/zephyr/spinlock.h

```diff
@@ -261,6 +261,20 @@ static ALWAYS_INLINE void k_spin_unlock(struct k_spinlock *l,
 	arch_irq_unlock(key.key);
 }
 
+/**
+ * @brief Checks if spinlock is held by some CPU, including the local CPU.
+ *		This API shouldn't be used outside the tests for spinlock
+ *
+ * @param l A pointer to the spinlock
+ * @retval true - if spinlock is held by some CPU; false - otherwise
+ */
+#if defined(CONFIG_SMP) && defined(CONFIG_TEST)
```

Comment:
This is not properly documented since k_spin_is_locked symbol is not set when building doxygen documentation.  The comment block ends up applying to K_SPINLOCK_BREAK further down this file.

![image](https://github.com/zephyrproject-rtos/zephyr/assets/128251/f1b3821b-ce15-4ea3-b809-7b640d111a35)

Option 1:

```suggestion
#if defined(CONFIG_SMP) && defined(CONFIG_TEST) || defined(__DOXYGEN__)
```

Option 2:

> k_spin_is_locked should be used for the spinlock testing only, so the scope of this API is intentionally restricted.

Should this be internal then?

Reply:
Hi @kartben , nice catch, thanks! I think both your suggestions make sense